calabash-cucumber/lib/calabash-cucumber/store/preferences.rb

```diff
@@ -13,14 +14,80 @@ def initialize
         @path = File.join(dot_dir, "preferences", "preferences.json")
       end
 
+      # !@visibility private
+      def usage_tracking
+        preferences = read
+
+        unless valid_user_tracking_value?(preferences[:usage_tracking])
+          log_defaults_reset
+          preferences[:usage_tracking] = defaults[:usage_tracking]
+          write(preferences)
```

Comment:
Good question.  If the preferences were an object file that was not editable by hand, I would agree: we should not write.

However, the file is a .json file which could be edited by hand.  This is a problem.  We cannot know if any given key is valid and I don't want to have to handle bad keys in the caller of this method.

> can't we just have defaults if unset?

We could, but if we are going to do that, it makes sense to persist those values to disk.

The .json file should be source of truth.

Imagine trying to debug this.  You see some odd behavior when calling the `usage_tracking` method.  You go to the .json file and you don't see a key for `:usage_tracking` or you see a value like `"all"` or `"ip"`.  Where do you go from there? You have to dig back through the source to find the defaults.